data_steward/validation/main.py

```diff
@@ -882,53 +881,47 @@ def _is_string_excluded_file(gcs_file_name):
+
     try:
-        storage_client = StorageClient()
         hpo_bucket = storage_client.get_hpo_bucket(hpo_id)
-        drc_private_bucket = storage_client.get_drc_bucket()
-        source_bucket = storage_client.get_bucket(hpo_bucket)
-        destination_bucket = storage_client.get_bucket(drc_private_bucket)
-        bucket_items = list_bucket(hpo_bucket)
-
-        ignored_items = 0
-        filtered_bucket_items = []
-        for item in bucket_items:
-            item_root = item['name'].split('/')[0] + '/'
-            if item_root.lower() in common.IGNORE_DIRECTORIES:
-                ignored_items += 1
-            else:
-                filtered_bucket_items.append(item)
+        drc_bucket = storage_client.get_drc_bucket()
+        bucket_items: list = storage_client.get_bucket_items_metadata(
+            hpo_bucket)
+    except BucketNotSet:
+        logging.info(f"Bucket for hpo_id '{hpo_id}' is empty/unset")
+    except BucketDoesNotExistError:  #TODO phase out BDNE error?  maybe?
+        logging.warning(f"Bucket '{hpo_id}' configured.")
+
+    # Filter
+    ignored_count: int = 0
+    allowed_items: list = []
+    for item in bucket_items:
+        item_root: str = item['name'].split('/')[0] + '/'
+        if item_root.lower() in common.IGNORE_DIRECTORIES:
+            ignored_count += 1
+        else:
+            allowed_items.append(item)
 
-        logging.info(f"Ignoring {ignored_items} items in {hpo_bucket}")
+    if ignored_count:
+        logging.info(
+            f"Ignoring {ignored_count} items in {storage_client._get_hpo_bucket_id(hpo_id)}"
+        )
 
-        prefix = f'{hpo_id}/{hpo_bucket}/'
+    prefix: str = f'{hpo_id}/{hpo_bucket.name}/'
 
-        for item in filtered_bucket_items:
-            item_name = item['name']
-            source_blob = source_bucket.get_blob(item_name)
-            destination_blob_name = f'{prefix}{item_name}'
-            source_bucket.copy_blob(source_blob, destination_bucket,
-                                    destination_blob_name)
-    except BucketDoesNotExistError as bucket_error:
-        bucket = bucket_error.bucket
-        # App engine converts an env var set but left empty to be the string 'None'
-        if bucket and bucket.lower() != 'none':
-            logging.warning(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' does not exist"
-            )
-        else:
-            logging.info(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' is empty/unset"
-            )
-    except HttpError as http_error:
-        message = (
-            f"Failed to copy files for hpo_id '{hpo_id}' due to the following "
-            f"HTTP error: {http_error.content.decode()}")
-        logging.exception(message)
+    # Copy
+    for item in allowed_items:
+        name = item['name']
+        full_name: str = f'{prefix}{name}'
+        hpo_blob = hpo_bucket.get_blob(name)
+        hpo_bucket.copy_blob(hpo_blob, drc_bucket, full_name)
```

Comment:
I'd suggest a log here with the count.
```suggestion
        hpo_bucket.copy_blob(hpo_blob, drc_bucket, full_name)
    logging.info(f'Copied {len(allowed_items)} items for bucket {storage_client._get_hpo_bucket_id(hpo_id)} for hpo_id {hpo_id}')
```

Reply:
The suggestion that Luis had has the ignore last, but instead appending to a list, we just upload directly

Reply:
Yeah that's a good suggestion. I'd suggest logging the allowed and ignored items in one log statement after.

Reply:
Yes.  the function will always provide a log.info with an "ignored" of "total" message

Reply:
Sounds good. From the total, we can infer the number of copied items